packages/mds-cache/index.ts

```diff
@@ -175,33 +189,57 @@ async function writeDeviceState(field: UUID, data: StateEntry) {
   return (await getClient()).hsetAsync(decorateKey('device:state'), field, JSON.stringify(data))
 }
 
-async function readTripsEvents(field: UUID): Promise<TripsEvents | null> {
-  const tripsEvents = await hget(decorateKey('trips:events'), field)
-  return tripsEvents ? parseTripsEvents(tripsEvents as StringifiedTripsEvents) : null
+async function readTripEvents(field: UUID): Promise<TripEvent[] | null> {
+  const tripEvents = await hget(decorateKey('trips:events'), field)
+  return tripEvents ? parseTripEvents(tripEvents as StringifiedTripEvents) : null
 }
 
-async function readAllTripsEvents(): Promise<{ [vehicle_id: string]: TripsEvents } | null> {
+// TODO: Investigate alternatives, this should be used temporarily. O(n) runtime for streaming is not ideal.
+async function readDeviceTripsEvents(pattern: string): Promise<TripsEvents | null> {
+  // hscan returns list of alternating key and value strings (i.e. [keyA, valueA, keyB, valueB])
+  const allFilteredTripEventsList = await hscan('trips:events', pattern)
+  if (allFilteredTripEventsList) {
+    let lastKey = ''
+    const allFilteredTripEvents = allFilteredTripEventsList.reduce((acc, entry, index) => {
+      if (index % 2 === 0) {
```

Comment:
Mind explaining exactly what's going on in this loop? The `{} as StringifiedAllTripsEvents` worries me a bit. From my parsing of how this would work, it looks like you're using the `entries` as `keys` in this map, when for the keys for this StringifiedAllTripsEvents should be `{ [vehicle_id: string]: StringifiedTripsEvents }`.

Reply:
With the hscan call we get a list of alternating key/value strings (i.e. [keyA, valueA, keyB, valueB]). It gross, but to maintain consistency with the input of the unflattener methods, where the actual parsing occurs, we transform this into an object of StringifiedAllTripsEvents. I want to look into alternatives to using this redis method, or ways to have it return a sensible format.

Reply:
Gotcha. Main modification I'd make here then is that you don't need to store `lastKey`, you can just do `allFilteredTripEventsList[index - 1]` to get the key, (and set that to a variable `key` or something) and you also can probably remove the `acc[entry] = ''` bit. Rough rewrite would probably look something like:

```typescript
allFilteredTripEventsList.reduce((acc: StringifiedTripsEvents, entry, index) => {
  if (index % 2 === 1) {
    const key = allFilteredTripEventsList[index - 1]
    acc[key] = entry
  }
  return acc
}, {})
```

Reply:
Also probably add a quick comment elaborating so future viewers aren't as confused as I was